### PR TITLE
Integration job

### PR DESCRIPTION
**Pull request by hornmo**

### CODE REVIEW

File: server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy [hunk 1]
```diff
@@ -1729,19 +1732,48 @@ class WorkflowController {
...
+          def old_from_combos = Combo.executeQuery("from Combo where fromComponent = ?", [otd])
+          def old_to_combos = Combo.executeQuery("from Combo where toComponent = ?", [otd])
+
+          old_from_combos.each { oc ->
+            def ctype = oc.type
+            def other_end = oc.toComponent
+            def existing_new = Combo.executeQuery("from Combo where type = ? and fromComponent = ? and toComponent = ?",[ctype, neworg, oc])
+
+            if (existing_new?.size() == 0) {
+              ofc.fromComponent = neworg
```
Comment (philboeselager) [inline server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy]: should be `oc.fromComponent = neworg` ?

**Review by philboeselager — APPROVED** [review]
Looks fine to me, just 1 comment, please see below.

If possible / suitable, smaller size of commits would be great, with a more concrete commit message. That would increase the intelligibility. (With respect to https://github.com/openlibraryenvironment/gokb/pull/9/commits/3ea67312566003e2a74a1a4bcfaf160408c21151.)

File: server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy [hunk 2]
```diff
@@ -756,235 +756,278 @@ class IntegrationController {
   @Secured(['ROLE_API', 'IS_AUTHENTICATED_FULLY'])
```
Comment (philboeselager) [inline server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy:756]: We should divide this huge 250 lines method into more intelligible and dealable submethods soon.

File: server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy (hunk 1 above)
Comment (hornmo) [reply server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy]: You are right, good catch!

File: server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy (hunk 2 above)
Comment (hornmo) [reply server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy:756]: That is a good idea.

**Pull request merged**